--- a/package/mtk/applications/luci-app-turboacc-mtk/Makefile
+++ b/package/mtk/applications/luci-app-turboacc-mtk/Makefile
@@ -8,7 +8,7 @@
 PKG_LICENSE:=GPL-3.0-only
 
 LUCI_TITLE:=LuCI support for MTK HNAT
-LUCI_DEPENDS:=+kmod-tcp-bbr @!PACKAGE_luci-app-turboacc
+LUCI_DEPENDS:=+kmod-tcp-bbr +kmod-nft-fullcone @!PACKAGE_luci-app-turboacc
 LUCI_PKGARCH:=all
 
 include $(TOPDIR)/feeds/luci/luci.mk
--- a/package/mtk/applications/luci-app-turboacc-mtk/htdocs/luci-static/resources/view/turboacc.js
+++ b/package/mtk/applications/luci-app-turboacc-mtk/htdocs/luci-static/resources/view/turboacc.js
@@ -161,6 +161,10 @@
 
 		s = m.section(form.NamedSection, 'config', 'turboacc');
 
+		o = s.option(form.Flag, 'fullcone', _('Full Cone NAT'));
+		o.default = o.disabled;
+		o.rmempty = false;
+		
 		o = s.option(form.ListValue, 'fastpath', _('Fastpath engine'),
 			_('The offloading engine for routing/NAT.'));
 		o.value('disabled', _('Disable'));
--- a/package/mtk/applications/luci-app-turboacc-mtk/root/etc/init.d/turboacc
+++ b/package/mtk/applications/luci-app-turboacc-mtk/root/etc/init.d/turboacc
@@ -68,4 +68,17 @@
 	local tcpcca
 	config_get "tcpcca" "config" "tcpcca" "cubic"
 	sysctl -w net.ipv4.tcp_congestion_control="$tcpcca" >"/dev/null"
+	
+	local fullcone=$(uci -q get turboacc.config.fullcone)
+	local current_fullcone=$(uci -q get firewall.@defaults[0].fullcone)
+	
+	if [ "$fullcone" != "$current_fullcone" ]; then
+		if [ "$fullcone" = "1" ]; then
+			uci set firewall.@defaults[0].fullcone='1'
+		else
+			uci delete firewall.@defaults[0].fullcone 2>/dev/null
+		fi
+		uci commit firewall
+		/etc/init.d/firewall restart
+	fi
 }
--- a/package/mtk/applications/luci-app-turboacc-mtk/root/etc/uci-defaults/turboacc
+++ b/package/mtk/applications/luci-app-turboacc-mtk/root/etc/uci-defaults/turboacc
@@ -43,6 +43,7 @@
 esac
 
 set_default tcpcca cubic
+set_default fullcone 1
 
 uci -q commit "turboacc"
 
